File: August-LeetCoding-Challenge/03-Valid-Palindrome/Valid-Palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int begin = 0;
        int end = s.Length-1;
        while(begin <= end){
            while(begin<=end && !Char.IsLetterOrDigit(s[begin])){
                begin++;
            }
            while(begin<=end && !Char.IsLetterOrDigit(s[end])){
                end--;
            }
            if(begin<=end && Char.ToLower(s[begin]) != Char.ToLower(s[end])){
                return false;
            }
            begin++;
            end--;
        }
        return true;
    }
}

File: August-LeetCoding-Challenge/03-Valid-Palindrome/Valid-Palindrome.kt
package string_integer

class ValidPalindromeKotlin125 {
    fun isPalindrome(s: String): Boolean {
        var left = 0
        var right = s.length - 1
        while (left < right) {
            while (left < right && !s[left].isLetterOrDigit()) {
                ++left
            }
            while (left < right && !s[right].isLetterOrDigit()) {
                --right
            }
            if (s[left++].toUpperCase() != s[right--].toUpperCase()) {
                return false
            }
        }
        return true
    }
}

fun main() {
    val solution = ValidPalindromeKotlin125()
    // true
    println(solution.isPalindrome("A man, a plan, a canal: Panama"))
    // false
    println(solution.isPalindrome("0P"))
}